File: backend/src/main/java/com/tribly/dto/device/response/DeviceRideDto.java
package com.tribly.dto.device.response;

import com.tribly.dto.validation.ValidateSchema;
import java.time.Instant;
import java.util.List;
import lombok.Builder;
import org.eclipse.microprofile.openapi.annotations.media.Schema;

@Schema(description = "Ride information for device applications")
@Builder
@ValidateSchema
public record DeviceRideDto(
    @Schema(description = "Team slug", required = true) String teamSlug,
    @Schema(description = "Ride slug", required = true) String rideSlug,
    @Schema(description = "Ride name", required = true) String rideName,
    @Schema(description = "Start date/time") Instant startDateTime,
    @Schema(description = "Route entries for this ride", required = true)
        List<DeviceRideEntryDto> entries) {}

File: frontend/src/api/dto/deviceRideDto.ts
import type { DeviceRideEntryDto } from './deviceRideEntryDto'
import type { Instant } from './instant'

/**
 * Ride information for device applications
 */
export interface DeviceRideDto {
  /** Team slug */
  teamSlug: string
  /** Ride slug */
  rideSlug: string
  /** Ride name */
  rideName: string
  /** Start date/time */
  startDateTime?: Instant
  /** Route entries for this ride */
  entries: DeviceRideEntryDto[]
}
